File: src/constants/formations.ts
import type { Formation, PlayFormat } from '../types';

// Standardformasjoner basert på NFFs retningslinjer for barnefotball.
// Hold én anbefalt per format — brukeren kan lage egendefinerte om ønsket.
export const DEFAULT_FORMATIONS: Formation[] = [
  {
    id: '5v5-2-2',
    name: '2-2',
    format: '5v5',
    positions: ['Keeper', 'V. forsvar', 'H. forsvar', 'V. angrep', 'H. angrep'],
    isCustom: false,
  },
  {
    id: '7v7-2-3-1',
    name: '2-3-1',
    format: '7v7',
    positions: ['Keeper', 'V. forsvar', 'H. forsvar', 'V. midtbane', 'Midtbane', 'H. midtbane', 'Spiss'],
    isCustom: false,
  },
  {
    id: '9v9-3-3-2',
    name: '3-3-2',
    format: '9v9',
    positions: ['Keeper', 'V. back', 'Midtstopper', 'H. back', 'V. midtbane', 'S. midtbane', 'H. midtbane', 'V. angrep', 'H. angrep'],
    isCustom: false,
  },
];

export function getFormationsForFormat(format: PlayFormat): Formation[] {
  return DEFAULT_FORMATIONS.filter(f => f.format === format);
}

export function getDefaultFormation(format: PlayFormat): Formation {
  return DEFAULT_FORMATIONS.find(f => f.format === format)!;
}

export function getFieldCount(format: PlayFormat): number {
  return getDefaultFormation(format).positions.length;
}

File: ios/ByttehjelpenKit/Sources/ByttehjelpenKit/PlanLogic/Formations.swift
import Foundation

/// Standardformasjoner basert på NFFs retningslinjer for barnefotball.
/// Speiler `src/constants/formations.ts` på web.
public enum DefaultFormations {
    public static let all: [FormationSnapshot] = [
        FormationSnapshot(
            id: UUID(uuidString: "00000000-0000-0000-0000-000000005a55")!,
            name: "2-2",
            format: .fiveOnFive,
            positions: ["Keeper", "V. forsvar", "H. forsvar", "V. angrep", "H. angrep"],
            isCustom: false
        ),
        FormationSnapshot(
            id: UUID(uuidString: "00000000-0000-0000-0000-000000007a77")!,
            name: "2-3-1",
            format: .sevenOnSeven,
            positions: ["Keeper", "V. forsvar", "H. forsvar", "V. midtbane", "Midtbane", "H. midtbane", "Spiss"],
            isCustom: false
        ),
        FormationSnapshot(
            id: UUID(uuidString: "00000000-0000-0000-0000-000000009a99")!,
            name: "3-3-2",
            format: .nineOnNine,
            positions: ["Keeper", "V. back", "Midtstopper", "H. back", "V. midtbane", "S. midtbane", "H. midtbane", "V. angrep", "H. angrep"],
            isCustom: false
        )
    ]

    public static func formations(for format: PlayFormat) -> [FormationSnapshot] {
        all.filter { $0.format == format }
    }

    public static func defaultFormation(for format: PlayFormat) -> FormationSnapshot {
        // Garantert ikke-tom siden vi har én standard per format.
        formations(for: format).first!
    }
}
